apps/src/templates/sessions/LockoutPanel.jsx

```diff
@@ -0,0 +1,283 @@
+              >
+                <strong>{i18n.sessionLockoutParentEmailField()}</strong>
+              </label>
+
+              {/* Slightly complicated layout allows for text underneath. */}
+              <div style={styles.fieldSection}>
+                <input
+                  style={styles.field}
+                  onChange={this.onEmailUpdate}
+                  onInput={this.onEmailUpdate}
+                  onBlur={this.onEmailUpdate}
+                  defaultValue={this.props.pendingEmail}
+                  id="parent-email"
+                />
+
+                {/* Show a 'Last email sent' prompt when available. */}
+                {pendingEmail && (
+                  <p style={styles.lastEmail}>
+                    <em>
+                      Last email sent:{' '}
+                      {requestDate.toLocaleDateString(locale, {
+                        ...dateOptions,
+                        hour: 'numeric',
+                        minute: 'numeric',
+                      })}
+                    </em>
+                  </p>
+                )}
+              </div>
+            </div>
+          </div>
+
+          {/* A sign-out button. */}
+          <Button
+            style={{...styles.button, float: isRTL ? 'left' : 'none'}}
+            text={i18n.signOutButton()}
+            color={Button.ButtonColor.gray}
+            href="/users/sign_out"
+          />
+
+          {/* The submit button. */}
+          {/* An empty onClick will still submit the form. */}
+          <Button
+            style={{...styles.button, float: isRTL ? 'none' : 'right'}}
+            text={
+              pendingEmail
+                ? i18n.sessionLockoutUpdateSubmit()
+                : i18n.sessionLockoutSubmit()
+            }
+            disabled={this.state.disabled}
+            onClick={() => {}}
+          />
+        </form>
+      </div>
+    );
+  }
+}
+
+const styles = {
+  container: {
+    border: '1px solid rgb(233 233 233)',
+    boxShadow: '5px 5px 3px 0px #ccc',
+    boxSizing: 'border-box',
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    marginTop: 10,
+    maxWidth: 700,
+    padding: 20,
+    paddingBottom: 0,
+  },
+  sections: {
+    display: 'table',
+    marginBottom: 25,
+  },
+  statusSection: {
+    marginBottom: 20,
+    marginTop: 20,
+  },
+  section: {
+    display: 'table-row',
+  },
+  notSubmitted: {
+    color: color.red,
+  },
+  pending: {
+    color: color.orange,
+  },
+  statusLabel: {
+    display: 'inline',
+    marginRight: 15,
+  },
+  statusLabelRTL: {
+    display: 'inline',
+    marginLeft: 15,
+  },
+  resendLink: {
+    fontSize: 'inherit',
+    textDecoration: 'underline',
+  },
+  label: {
+    display: 'table-cell',
+    paddingRight: 15,
+  },
+  labelRTL: {
+    display: 'table-cell',
+    paddingLeft: 15,
+  },
+  field: {
+    color: '#292F36',
+    border: '1px solid #C6CACD',
+    borderRadius: 4,
+    padding: 6,
+  },
+  fieldSection: {
+    display: 'table-cell',
```

Comment:
My understanding is that it's better to use [Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for layout stuff instead of trying to use tables or table styling. Did you already try using it?

Reply:
That's kinda true. I could try to use a full width flex box for that section and grow the field side. I think if I establish the flex basis, it won't squish the text. Hmm. Yes, I think that would do the same thing.

Reply:
Gosh expanding `<input>` tags to the full width of grown divs in a flexbox is silly annoying, but I have done it... will test it a little and push it up.

Reply:
Done.